sentry_sdk/integrations/excepthook.py

```diff
@@ -19,14 +20,15 @@ def install(self):
 
 def _make_excepthook(old_excepthook):
     def sentry_sdk_excepthook(exctype, value, traceback):
-        with capture_internal_exceptions():
-            hub = Hub.current
-            event, hint = event_from_exception(
-                (exctype, value, traceback),
-                with_locals=hub.client.options["with_locals"],
-                mechanism={"type": "excepthook", "handled": False},
-            )
-            hub.capture_event(event, hint=hint)
+        if ExcepthookIntegration.current is not None:
+            with capture_internal_exceptions():
+                hub = Hub.current
+                event, hint = event_from_exception(
+                    (exctype, value, traceback),
+                    with_locals=hub.client.options["with_locals"],
```

Comment:
I'm mostly concerned because the thing is nullable and every time I read such code I'm not sure if the programmer validated that it's not none here. Rust broke me to look at nullables in a different way now :) Performance i'm not that much concerned.

Reply:
Ok, I know what you mean but this instance of nullability in particular should be caught by tests. I don't think adding another layer of indirection is really going to help.